### PR TITLE
[Sparse] C&S Example

### PR DESCRIPTION
**Pull request by mufeili**
## Description
Accuracy is about 0.74~0.75

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/sparse/c_and_s.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    def __init__(self, num_layers=20, alpha=0.9):
+        super().__init__()
+
+        self.num_layers = num_layers
+        self.alpha = alpha
+
+    ###########################################################################
+    # (HIGHLIGHT) Compute Label Propagation with Sparse Matrix API
+    ###########################################################################
+    @torch.no_grad()
```
Comment (jermainewang) [inline examples/sparse/c_and_s.py:24]: Shall we make this a preprocessing step like SIGN?

Comment (mufeili) [reply examples/sparse/c_and_s.py:24]: That's definitely feasible. A follow-up question is weather we still want to have a `CorrectAndSmooth` class or simply two functions `correct` and `smooth`.

File: examples/sparse/c_and_s.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        print(
+            f"Base model, In epoch {epoch}, loss: {loss:.3f}, "
+            f"val acc: {val_acc:.3f}, test acc: {test_acc:.3f}"
+        )
+
+    # Stage2: Correct and Smooth.
+    soft_label = F.softmax(logits, dim=1)
+    label = F.one_hot(label)
+    soft_label = model.correct(A_hat, label, soft_label, train_mask)
+    soft_label = model.smooth(A_hat, label, soft_label, train_mask)
```
Comment (jermainewang) [inline examples/sparse/c_and_s.py:109]: Let's write them as plain functions:
```
soft_label = correct(A_hat, label, soft_label, train_mask)
soft_label = smooth(A_hat, label, soft_label, train_mask)
```

In this way, you don't need to pass the `model` object anymore and you can rename `base_model` to `model`.

File: examples/sparse/c_and_s.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    # Create models.
+    X = g.ndata["feat"]
+    in_size = X.shape[1]
+    out_size = dataset.num_classes
+    base_model = nn.Linear(in_size, out_size).to(dev)
+    model = CorrectAndSmooth().to(dev)
+
+    # Kick off training.
+    train(base_model, model, g, A_hat, X)
```
Comment (jermainewang) [inline examples/sparse/c_and_s.py:143]: How about we further split c&s with `train`? This makes it very clear that the base models are trained without graph while the post-process step needs graph.

**Review by frozenbugs — APPROVED** [review]
